tools/ci/identical_variables.py

```diff
@@ -0,0 +1,66 @@
+from avulto import DME
+from collections import namedtuple
+import os
+import sys
+import time
+
+Failure = namedtuple("Failure", ["filename", "lineno", "message"])
+
+RED = "\033[0;31m"
+GREEN = "\033[0;32m"
+BLUE = "\033[0;34m"
+NC = "\033[0m"  # No Color
+
+def print_error(message: str, filename: str, line_number: int):
+    if os.getenv("GITHUB_ACTIONS") == "true": # We're on github, output in a special format.
+        print(f"::error file={filename},line={line_number},title=Identical Variables::{filename}:{line_number}: {message}")
+    else:
+        print(f"{filename}:{line_number}: {message}")
+
+def main():
+    print("identical_variables started")
+    exit_code = 0
+    start = time.time()
+
+    dme = DME.from_file("paradise.dme")
+    all_failures = []
+
+    for path in dme.subtypesof("/"):
+        typepath = dme.type_decl(path)
+
+        for variable_name in typepath.var_names(modified=True):
+            modded = typepath.var_decl(variable_name, False)
+            if(modded == None):
+                print("Something went wrong!!!")
+                sys.exit(1)
+                return
+            if path.parent == "/":
+                continue
+            parent_typepath = dme.type_decl(path.parent)
+            original = parent_typepath.var_decl(variable_name, True)
+            if(modded.const_val == original.const_val):
+                if(modded.const_val == None): # Both proc calls (like sound() or icon()) and nulls are treated as "None", this sucks.
+                    continue
```

Comment:
yeah, this is kind of fiddly. i didn't want to expose AST nodes to this part of the API (everything in a const_val is a pythonic conversion of its DM equivalent) and if the declaration's value is a non constant expression that's really the only sensible thing to return, possibly in a 'non_const_expr' attribute or similar. but that does mean there's no way to distinguish between a declaration whose const value is null and one whose declaration is a non-const expression. patches accepted 😇